### ISSUE TITLE
Rule ignored when in node environment

### ISSUE THREAD

**Issue by backflip** [eslint/eslint#1108]
After reading "Configuration Cascading and Hierarchy" in the docs I was under the impression that the code example below should not report an error. What am I missing?

.eslintrc

``` json
{
    "rules": {
        "no-mixed-requires":0
    }
}
```

gulpfile.js

``` js
/* eslint-env node */

"use strict";

var gulp = require("gulp"),
    a = 1;

gulp.task("default", function() {
    console.log(a);
});
```

Result of `eslint gulpfile.js`:

``` bash
gulpfile.js
  5:0  error  Do not mix 'require' and other declarations  no-mixed-requires

✖ 1 problem
```
